blocks/location-overview/location-overview.js

```diff
@@ -0,0 +1,94 @@
+import {
+  domEl,
+  div,
+  a,
+  span,
+  h2,
+  p,
+} from '../../scripts/dom-helpers.js';
+
+import {
+  decorateIcons,
+} from '../../scripts/aem.js';
+
+function getLearnMoreIcon() {
+  const learnMoreIcon = domEl('calcite-icon', {
+    icon: 'arrowRight',
+    scale: 's',
+  });
+
+  const iconContainer = div({ class: 'learn-more-icon-container' }, 'Learn more', learnMoreIcon);
+  return iconContainer;
+}
+
+function getPanelIcon(iconPath = '') {
+  const panelIcon = span({
+    class: iconPath || 'location-overview-panel-icon icon icon-2-5d-mesh-48',
+  });
+
+  return panelIcon;
+}
+
+function getContainerContents(hText, pText, panelIconPath) {
+  const panelIcon = getPanelIcon(panelIconPath);
+  const lernMoreIcon = getLearnMoreIcon();
+
+  const overviewCardHeader = h2({
+    class: 'location-overview-card-header',
```

Comment:
Is this class needed? isn't this the only h2 anyway?

Reply:
Yes, it is used here: https://github.com/aemsites/esri/pull/124/files#diff-5c40cda8c14907ead8c5d3569ae50810ab87fba2c9a892341239e54324e59121R97


Default h2 is not inline with what we want there.

Reply:
I understand you're using it there, but can't you use `.location-overview h2` instead?

Reply:
Sure I can,  but that introduces an artificial coupling we don't necessarily want.

This way we have a flat selection of just one class.

The way you propose is quite the opposite of this idea of composition in JavaScript from the ground up.

We would use those selection methods if we wouldn't build the nodes ourselves, but use it the way it got imported.

Reply:
While it's ok to make changes, it tends to be simpler to avoid making unnecessary changes to the DOM or add extra classes. A heading having a special styling is something natural and to be expected within a block, what value does the extra class add?

Reply:
> A heading having a special styling is something natural and to be expected within a block, what value does the extra class add?

Exactly that. What if you add another heading?

If you select <parent> <child> you will need to select <parent> <nthchild> in case you want to style them differently. Adding a class is decoupling these elements, giving you flexibility.

Reply:
In general we don't really care too much about coupling between the js and css file in the same block, as they're easy to change together, and you have to also weigh it vs the added complexity you're introducing by adding extra logic to the js file. In general you'd expect all h2 within a block to look the same, rather than each of them having different styles.